unlock-app/src/components/interface/checkout/main/Confirm/ConfirmUniversalCard.tsx

```diff
@@ -0,0 +1,234 @@
+import { useActor } from '@xstate/react'
+import { storage } from '~/config/storage'
+
+import { PoweredByUnlock } from '../../PoweredByUnlock'
+import { ViewContract } from '../../ViewContract'
+import { PricingData } from './PricingData'
+import { CryptoElements, OnrampElement } from '../../utils/CryptoElements'
+import { CreditCardPricingBreakdown } from './ConfirmCard'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useUniversalCardPrice } from '~/hooks/useUniversalCardPrice'
+import { usePurchaseData } from '~/hooks/usePurchaseData'
+
+interface Props {
+  injectedProvider: unknown
+  checkoutService: CheckoutService
+  onConfirmed: (lock: string, hash?: string) => void
+  onError: (message: string) => void
+}
+
+export function ConfirmUniversalCard({
+  injectedProvider,
+  checkoutService,
+  onConfirmed,
+  onError,
+}: Props) {
+  const { getWalletService, account } = useAuth()
+  const [state] = useActor(checkoutService)
+  const config = useConfig()
+  const [sessionError, setSessionError] = useState<string>('')
+  const [onrampSession, setOnrampSession] = useState<any>(null)
+  const stripeOnrampPromise = loadStripeOnramp(config.stripeApiKey)
+
+  const { lock, recipients, captcha, paywallConfig, password, promo } =
+    state.context
+
+  // Build the `purchaseData` field that gets passed to the contract
+  const { isInitialLoading: isInitialDataLoading, data: purchaseData } =
+    usePurchaseData({
+      lockAddress: lock!.address,
+      network: lock!.network,
+      promo,
+      password,
+      captcha,
+      paywallConfig,
+      recipients,
+    })
+
+  // And now get the price to pay by card
+  const { data: cardPricing, isInitialLoading: isCardPricingLoading } =
+    useUniversalCardPrice({
+      network: lock!.network,
+      lockAddress: lock!.address,
+      recipients,
+      purchaseData: purchaseData!,
+      enabled: !isInitialDataLoading,
+    })
+
+  // TODO: Also configure webhook on the Stripe side?
+  const onChange = async (event: any) => {
+    const { session } = event
+
+    if (
+      session?.wallet_address &&
+      account &&
+      session?.wallet_address.toLowerCase() !== account.toLowerCase()
+    ) {
+      setSessionError(
+        'You cannot change the recipient address as it is the address that will receive the NFT membership token. Please start again.'
+      )
+    }
+
+    const expectedAmount = cardPricing?.total
+    if (
+      session.quote.destination_amount &&
+      100 * parseFloat(session.quote.destination_amount) !== expectedAmount
+    ) {
+      console.error(
+        'Price changed',
+        100 * parseFloat(session.quote.destination_amount),
+        expectedAmount
+      )
+      setSessionError('You cannot change the amount. Please start again.')
+    }
+
+    if (session.status === 'fulfillment_complete') {
+      const response = await storage.captureOnRampSession(
+        session.id,
+        session.quote.blockchain_tx_id
+      )
```

Comment:
Could we wrap this in useQuery so we can track errors and retry in bad network condition automatically?